src/modules/track_model/src/track_model.cc

```diff
@@ -1 +1,298 @@
-#include "track_model.h"
\ No newline at end of file
+#include "track_model.h"
+
+#include <iostream>
+#include <string>
+#include <random>
+
+#include "train_model.h"
+#include "types.h"
+#include "block_builder.h"
+#include "graph.h"
+
+namespace track_model
+{
+
+//constructor for when a track is passsed in
+types::Error SoftwareTrackModel::SetTrackLayout(const types::TrackId track, const std::vector<types::Block> &blocks)
+{
+    //setting trackID
+    track_ = track;
+
+    //fill vector with blocks
+    blocks_ = blocks;
+
+    //TODO: FILL GRAPH WITH BLOCKS HERE
+
+    for (int i = 0; i < blocks_.size(); i++)
+    {
+        // every edge is really the length of the second block
+        graph.AddEdge(i, i + 1, blocks_[i + 1].length);
+    }
+
+    return types::ERROR_NONE;
+}
+
+types::TrackId SoftwareTrackModel::GetTrackId(void)
+{
+    // Implementation logic to get the Track ID
+    return track_;
+}
+
+types::Error SoftwareTrackModel::AddTrainModel(std::shared_ptr<train_model::TrainModel> train)
+{
+    // Implementation logic to add a train model
+
+    //TODO: ADD FAILURE STATES
+
+    //add model
+    trains_.push_back(train);
+
+    //populate occupied train block 2d vector with a new vector for this train
+    occupied_train_blocks_.push_back({0});
```

Comment:
How do you keep track of the index in each vector with the data for a particular train?  What happens when a train returns to the yard and leaves the track?